src/ui/control/navigation_control.js

```diff
@@ -35,13 +41,18 @@ class NavigationControl {
 
         this._zoomInButton = this._createButton('mapboxgl-ctrl-icon mapboxgl-ctrl-zoom-in', 'Zoom In', () => this._map.zoomIn());
         this._zoomOutButton = this._createButton('mapboxgl-ctrl-icon mapboxgl-ctrl-zoom-out', 'Zoom Out', () => this._map.zoomOut());
-        this._compass = this._createButton('mapboxgl-ctrl-icon mapboxgl-ctrl-compass', 'Reset North', () => this._map.resetNorth());
-        this._compassArrow = DOM.create('span', 'mapboxgl-ctrl-compass-arrow', this._compass);
+
+        if (this.options.showCompass) {
+            this._compass = this._createButton('mapboxgl-ctrl-icon mapboxgl-ctrl-compass', 'Reset North', () => this._map.resetNorth());
+            this._compassArrow = DOM.create('span', 'mapboxgl-ctrl-compass-arrow', this._compass);
+        }
```

Comment:
If the `this._compass` element is not created for `showCompass: false`, then we should also make sure in `onAdd()` not to create the `DragRotateHandler`, whose constructor accepts that element as an argument.

Reply:
I thought the same, but if no element it passed in the `DragRotateHandler` will default to the `canvasContainer` :)

Reply:
> if no element it passed in the DragRotateHandler will default to the canvasContainer

Ah, I see!  Good point...  but even so, I think we should avoid it: that default behavior of `DragRotateHandler` isn't documented as part of its internal API, and it could easily end up being removed in a refactor.  

> Especially not to break stuff that's already out there.

What do you see as potentially breaking?

Reply:
As @anandthakker suggests, we do need this change. In addition to the potential future compatibility issues, there is also the fact that the map creates and registers a `DragRotateHandler` of its own. If we let the navigation control create a second one bound to the same element, that will result in duplicate event handlers and probably produce unexpected behavior.